lips/lip-20.md

```diff
@@ -0,0 +1,595 @@
+resource that is uniquely published for each registered currency type `C` under the [Libra Root](https://github.com/libra/lip/blob/master/lips/lip-2.md#roles) (see also the implementation of
+[`Libra::register_SCS_currency`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-register_scs_currency)
+and
+[`Libra::register_currency`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-register_currency))
+account address. We now describe each of these data and their purpose.
+
+### <a name="to_lbr_exchange_rate">`to_lbr_exchange_rate`</a>
+
+There are certain cases where values in two different currencies need to be
+compared against each other. This exchange rate is used to normalize both values to their
+approximate ≋LBR value so that their values may be compared. In particular:
+* Transactions can set their gas price in different currencies, however transactions
+  need to be ranked against each other based upon their gas prices. This field is used by
+  the rest of the system to normalize gas prices to one single "unit" for comparison and
+  ranking purposes.
+* The dual attestation threshold needs to apply to payments in all currencies. This limit is set in
+  terms of ≋LBR, transfers are first normalized using this exchange rate and then compared
+  against the limit.
+
+#### Mutability
+The initial exchange rate to ≋LBR must be provided at the time of registration. The
+exchange rate may be [updated](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-update_lbr_exchange_rate)
+by an account with the [Treasury Compliance role](https://github.com/libra/lip/blob/master/lips/lip-2.md#roles) to correspond to
+fluctuations in the real-time exchange rate between the specified currency `C` and
+≋LBR. Note that this is not meant to be an exact exchange rate and _should not_ be used for
+determining exchange rates for value transfer between different currencies.
+
+### `is_synthetic`
+
+This defines whether the given currency is synthetic or an SCS. In the case of
+[`Libra::register_SCS_currency`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-register_scs_currency)
+the function sets this to `false`. For
+[`Libra::register_currency`](https://github.com/libra/libra/blob/master/language/stdlib/modules/doc/Libra.md#function-register_currency)
+this must be provided to the function.
+
+#### Mutability
+Immutable
+
+### <a name="scaling_factor">`scaling_factor`</a>
+
+This defines what fraction each unit of value on-chain represents in the off-chain
+representation of the currency. As an example: the on-chain representation of ≋LBR is
+microlibra, therefore each unit of value on-chain represents one microlibra and the
+`scaling_factor` for ≋LBR is `1,000,000` since each unit is `1/1,000,000` of an ≋LBR.
```

Comment:
This example is extremely helpful! But would be good to use the same dummy `C` type for consistency with the `fractional_part` example.